src/app/public/modules/datepicker/datepicker-input-fuzzy.directive.ts

```diff
@@ -340,7 +340,9 @@ export class SkyFuzzyDatepickerInputDirective
     this.onTouched();
 
     let formattedDate = this.fuzzyDateService.getStringFromFuzzyDate(this.value, this.dateFormat);
-    this.setInputElementValue(formattedDate || '');
+    if (this.control.valid) {
+      this.setInputElementValue(formattedDate || '');
```

Comment:
In the valid case - what is the point of having the `|| ''` here? If it is valid then there should always be a valid `formattedDate`, correct? Or am I missing something?

Reply:
The empty string allows end users to still clear the form out. If that wasn't there, we'd be sending a value of `undefined` to the `setInputElementValue()` method.

Reply:
Gotcha. LGTM